### ISSUE TITLE
Star Gladiator SLS Wrath/Anger only factor in mastery ATK in their formula

### ISSUE THREAD

**Issue by Dakado** [rathena/rathena#2823]
~~According to: https://github.com/rathena/rathena/pull/2811 this issue was fixed but I can confirm that the bug still somehow persists, because this skill:
http://ratemyserver.net/skill_db.php?skid=437&small=1&back=1
does not work at all on latest rAthena.
(I have used the source from #2811 even though the pull is not accepted yet, but I am experiencing the same behavior even on the eb1635f)~~

~~For testing purposes I have added + 20000 to the skill rate which is later by the way incorrectly calculated, according to the description of the skill it should be "skillratio /= 13 - 3 * skill;" not 12, but that is not the problem I am referencing now, even if I have added those 20000 my attack is still exactly the same (no dmg boost at all while attacking on star monter), no change at all, which proves that this snipped taken from battle.cpp is never called or never actively calculated into the final damage of each attack.~~

```
		if (i < MAX_PC_FEELHATE && (skill = pc_checkskill(sd, sg_info[i].anger_id))) {
			int skillratio = sd->status.base_level + sstatus->dex + sstatus->luk + 20000; //ADDED VALUE FOR TEST
			if (i == 2)
				skillratio += sstatus->str; //Star Anger
			if (skill < 4)
				skillratio /= 12 - 3 * skill;
			ATK_ADDRATE(wd.damage, wd.damage2, skillratio);
#ifdef RENEWAL
			ATK_ADDRATE(wd.masteryAtk, wd.masteryAtk2, skillratio);
#endif
		}
```

**Comment by Dakado** [comment]
UPDATE: After equiping any weapon the skill does not work, with no weapon equiped at all it works as intended.

**Comment by secretdataz** [comment]
Yes, the snippet is called.
![image](https://user-images.githubusercontent.com/2021243/35229999-769ee0b8-ffc7-11e7-9aa2-b9163e3f2d3c.png)

The reason why the skill doesn't have any effect with any weapon equipped is, currently SLS wrath skills only take effect on mastery ATK which SG, when equipped with a weapon, does not have any.
Instead, with bare hands, SG has 10*(TK_RUN level) mastery ATK which will be factored in SLS wrath formulae.

**Comment by Dakado** [comment]
@secretdataz Ok that makes sense, where can I edit that piece of code that defines that SG loses all ATK when equiped with weapon ? Can you please link me the snippet ? Because I remember that SG had always had bigger damage with books + miracle activated, I know I cannot prove it so I dont even ask for fix, I want to edit it just myself instead of wasting your + mine time, it would be better for both of us.

+ I dont know about the ATK_ADDRATE but I think it is also wrongly calculated, considering that I have BaseLV 255, STR 177 + 270, DEX 101 + 61, LUK 100 + 55 which is in total 1022, so diving it by 3 we should get ATK +340%, but after performing tests without any weapon I do get around 53% bonus in  damage output on dummy in prontera.

**Comment by secretdataz** [comment]
Closed in f2c7384
